blazar/plugins/networks/network_plugin.py

```diff
@@ -329,9 +329,8 @@ def on_start(self, resource_id):
     def delete_port(self, neutron, ironic, port):
         if port['binding:vnic_type'] == 'baremetal':
             node = port.get('binding:host_id')
-            if node:
-                ironic.node.vif_detach(node, port['id'])
```

Comment:
Can you explain why it's safe to not do this?

Reply:
I think this code may still be useful if one is using separate leases for network and nodes.

Reply:
Yea that makes since if the leases are separate. I added a check for if there is an instance on the node before attempting to detach the vif.